libc/calls/getprogramexecutablename.greg.c

```diff
@@ -79,6 +79,7 @@ static inline void GetProgramExecutableNameImpl(char *p, char *e) {
   // if argv[0] exists then turn it into an absolute path. we also try
   // adding a .com suffix since the ape auto-appends it when resolving
   if (__argc && (((q = __argv[0]) && !sys_faccessat(AT_FDCWD, q, F_OK, 0)) ||
+                 IsMetal() ||
```

Comment:
Wouldn't it be simpler to just add a statement higher up that says:

```c
if (IsMetal()) {
  strcpy(p, "/proc/self/exe");
  return;
}
```

Reply:
@jart : I have made the requested addition.  (Except, I am using `memccpy` rather than `strcpy`, in order to take the output buffer size into account.)  Thank you.